integration/smartbft/smartbft_block_deliverer_test.go

```diff
@@ -0,0 +1,205 @@
+	}
+
+	joinChannel(network, channel)
+
+	By("Waiting for followers to see the leader")
+	Eventually(ordererRunners[1].Err(), network.EventuallyTimeout, time.Second).Should(gbytes.Say("Message from 1"))
+	Eventually(ordererRunners[2].Err(), network.EventuallyTimeout, time.Second).Should(gbytes.Say("Message from 1"))
+	Eventually(ordererRunners[3].Err(), network.EventuallyTimeout, time.Second).Should(gbytes.Say("Message from 1"))
+
+	/* Create a stream with client for each orderer*/
+	var allStreams []ordererProtos.AtomicBroadcast_BroadcastClient
+	for _, o := range network.Orderers {
+		conn := network.OrdererClientConn(o)
+		broadcaster, err := ordererProtos.NewAtomicBroadcastClient(conn).Broadcast(context.Background())
+		Expect(err).ToNot(HaveOccurred())
+		allStreams = append(allStreams, broadcaster)
+	}
+
+	/* Fill the ledger with blocks */
+	By("Filling ledger with blocks")
+	closeTxProducer := make(chan bool)
+	go func() {
+		defer GinkgoRecover()
+		var i uint64
+		for {
+			buff := make([]byte, 8)
+			binary.BigEndian.PutUint64(buff, i)
+
+			/* Create transaction envelope which will be broadcast to all orderers */
+			env := ordererclient.CreateBroadcastEndorserTransactionEnvelope(network, network.Orderers[1], channel, buff)
+
+			for _, stream := range allStreams {
+				err := stream.Send(env)
+				Expect(err).NotTo(HaveOccurred())
+			}
+			i++
+
+			finishSending := false
+			select {
+			case <-closeTxProducer:
+				finishSending = true
+			case <-time.After(time.Second):
+			}
+			if finishSending {
+				return
+			}
+		}
+	}()
+	Eventually(ordererRunners[1].Err(), 10*time.Second).Should(gbytes.Say("Delivering proposal, writing block 10"))
+	closeTxProducer <- true
+
+	/* Shut down the orderers */
+
+	By("Taking down all the orderers")
+	for _, proc := range ordererProcesses {
+		proc.Signal(syscall.SIGINT)
+		Eventually(proc.Wait(), network.EventuallyTimeout).Should(Receive())
+	}
+
+	/* Extract ledger from one of the orderers */
+	By("Extracting blocks from ledger")
+	ordererFabricConfig := network.ReadOrdererConfig(network.Orderers[0])
+	blockStoreProvider, err := blkstorage.NewProvider(
+		blkstorage.NewConf(ordererFabricConfig.FileLedger.Location, -1),
+		&blkstorage.IndexConfig{
+			AttrsToIndex: []blkstorage.IndexableAttr{blkstorage.IndexableAttrBlockNum},
+		},
+		&disabled.Provider{},
+	)
+	Expect(err).NotTo(HaveOccurred())
+	blockStore, err := blockStoreProvider.Open(channel)
+	Expect(err).NotTo(HaveOccurred())
+	fileLedger := fileledger.NewFileLedger(blockStore)
+	var ledgerArray []*common.Block
+	for i := uint64(0); i < fileLedger.Height(); i++ {
+		block, err := fileLedger.RetrieveBlockByNumber(i)
+		Expect(err).NotTo(HaveOccurred())
+		ledgerArray = append(ledgerArray, block)
+	}
+	/* The ledger now saved and can be accessed at ledgerArray */
+
+	/* Create orderer mocks and make sure they can access the ledger */
+	var mocksArray [4]*MockOrderer
+
+	for i, orderer := range network.Orderers {
```

Comment:
we can decouple the number of orderers and mocks. We can generate the ledegr from 4 orderers but then span the ledger across N mock.